pkg/alerter/alert.go

```diff
@@ -1,8 +1,40 @@
 package alerter
 
+import "encoding/json"
+
 type Alerts map[string][]Alert
 
-type Alert struct {
-	Message              string `json:"message"`
-	ShouldIgnoreResource bool   `json:"-"`
+type Alert interface {
+	Message() string
+	ShouldIgnoreResource() bool
+}
+
+type SerializableAlert struct {
```

Comment:
It can't work, this struct is useless, you have to implement `UnmarshalJSON` and `MarshalJSON` for every struct that implement the Alert interface

Reply:
I have pushed new state of my pr, you can check if you want, but i didn't touch this conversation for the moment. I'll do it tomorrow.